scipy/optimize/_qap.py

```diff
@@ -0,0 +1,788 @@
+
+    .. math::
+
+        \min_P & \ {\ \text{trace}(A^T P B P^T)}\\
+        \mbox{s.t. } & {P \ \epsilon \ \mathcal{P}}\\
+
+    where :math:`\mathcal{P}` is the set of all permutation matrices,
+    and :math:`A` and :math:`B` are square matrices.
+
+    Graph matching tries to *maximize* the same objective function.
+    This algorithm can be thought of as finding the alignment of the
+    nodes of two graphs that minimizes the number of induced edge
+    disagreements, or, in the case of weighted graphs, the sum of squared
+    edge weight differences.
+
+    Note that the quadratic assignment problem is NP-hard, is not
+    known to be solvable in polynomial time, and is computationally
+    intractable. Therefore, the results given are approximations,
+    not guaranteed to be exact solutions.
+
+
+    Parameters
+    ----------
+    A : 2d-array, square
+        The square matrix :math:`A` in the objective function above.
+
+    B : 2d-array, square
+        The square matrix :math:`B` in the objective function above.
+
+    method :  str in {'faq', '2opt'} (default: 'faq')
+        The algorithm used to solve the problem.
+        :ref:`'faq' <optimize.qap-faq>` (default) and
+        :ref:`'2opt' <optimize.qap-2opt>` are available.
+
+    options : dict, optional
+        A dictionary of solver options. All solvers support the following:
+
+        partial_match : 2d-array of integers, optional, (default = None)
+            Allows the user to fix part of the matching between the two
+            matrices. In the literature, a partial match is also
+            known as a "seed" [2]_.
+
+            Each row of `partial_match` specifies the indices of a pair of
+            corresponding nodes, that is, node ``partial_match[i, 0]`` of `A`
+            is matched to node ``partial_match[i, 1]`` of `B`. Accordingly,
+            ``partial_match`` is an array of size ``(m , 2)``, where ``m`` is
+            not greater than the number of nodes.
+
+        maximize : bool (default = False)
+            Setting `maximize` to ``True`` solves the Graph Matching Problem
+            (GMP) rather than the Quadratic Assingnment Problem (QAP).
+
+        rng : {None, int, `~np.random.RandomState`, `~np.random.Generator`}
+            This parameter defines the object to use for drawing random
+            variates.
+            If `rng` is ``None`` the `~np.random.RandomState` singleton is
+            used.
+            If `rng` is an int, a new ``RandomState`` instance is used,
+            seeded with `rng`.
+            If `rng` is already a ``RandomState`` or ``Generator``
+            instance, then that object is used.
+            Default is None.
+
+        For method-specific options, see
+        :func:`show_options('quadratic_assignment') <show_options>`.
+
+    Returns
+    -------
+    res : OptimizeResult
+        A :class:`scipy.optimize.OptimizeResult` containing the following
```

Comment:
The other methods in `scipy.optimize` just use `OptimizeResult`. I think numpydoc takes care of the rest.